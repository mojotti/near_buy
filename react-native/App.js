import React from 'react';
import { Provider } from 'react-redux';
import Application from './components/Application';
import store from './redux';

export default class App extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <Application />
      </Provider>
    );
  }
}
