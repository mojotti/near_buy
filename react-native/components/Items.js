'use strict';

import React from 'react';
import { Button,
    ListView,
    StyleSheet,
    Text,
    View } from 'react-native';
import { logout } from '../redux/actions/auth';
import { connect } from 'react-redux';
import { localhost,
    width } from "../src/static/constants";

export class Items extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            dataSource: new ListView.DataSource({
                rowHasChanged: (row1, row2) => row1 !== row2,
            }),
            loaded: false,
            data: ''
        };
    }

    static navigationOptions = {
        title: 'Your items',
        headerLeft: null
    };

    userLogout() {
        this.props.onLogout();
    }

    componentDidMount() {
        this.fetchData();
    }

    getHeaders() {
        let headers = new Headers();
        headers.append('Authorization', `Bearer ${this.props.token}`);
        return headers;
    }

    fetchData() {
        let url = `http://${localhost}:5000/api/v1.0/user/items`;
        fetch(url, {
            method: 'GET',
            headers: this.getHeaders()
        })
        .then((response) => response.json())
        .then((responseJson) => {
            this.handleAllItemsRequest(responseJson);
        })
        .catch((error) => {
        console.error(error);
        });
    }

    handleAllItemsRequest(responseJson) {
        if (responseJson.items === 'no items') {
        this.setState({
            loaded: true,
            data: responseJson.items,
        });
        } else {
            this.setState({
                dataSource: this.state.dataSource.cloneWithRows(responseJson.items),
                loaded: true,
                data: responseJson.items,
            });
        }
    }

    navigateToNewItem() {
        const { navigate } = this.props.navigation;
        navigate('NewItem');
    }

    render() {
        return (
            <View style={[styles.container]}>
                {this.renderUserData()}
                <Button
                    title='Add new item'
                    onPress={() => this.navigateToNewItem()}
                />
                <Button
                    onPress={() => this.userLogout()}
                    title='Logout'
                    style = {{margin: 15}}
                />
            </View>
        );
    }

    renderUserData() {
        if (this.state.loaded === false) {
            return (
                <Text style={[styles.infoText]}>
                    {'Loading user data...'}
                </Text>
            )
        }
        if (this.state.data === 'no items') {
            return (
                <Text style={[styles.infoText]}>
                    {`You don't have any items yet. You should start to sell some shit!`}
                </Text>
            );
        } else {
            return (
                <ListView
                    dataSource={this.state.dataSource}
                    renderRow={data => this.renderItem(data)}
                />
            );
        }
    }

    renderItem(item) {
        return (
            <View style={[styles.item, styles.separator]}>
                <Text>
                    {`Title: ${item.title}`}
                </Text>
                <Text>
                    {`Description: ${item.description}`}
                </Text>
                <Text>
                    {`Price: ${item.price} â‚¬`}
                </Text>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
    },
    item: {
        padding: 10,
        backgroundColor: 'azure',
        width: width,
    },
    separator: {
        marginBottom: 1,
        borderBottomWidth: 1,
        borderBottomColor: 'gainsboro',
    },
    infoText: {
        textAlign: 'center',
        padding: 10
    }
});

const mapStateToProps = (state, ownProps) => {
    return {
        username: state.auth.username,
        token: state.auth.token
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        onLogout: () => { dispatch(logout()); }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Items);
