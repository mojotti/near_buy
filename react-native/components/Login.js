'use strict';

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Alert,
    Button,
    Image,
    KeyboardAvoidingView,
    StyleSheet,
    Text,
    TextInput,
    View } from 'react-native';
import { login } from '../redux/actions/auth';
import { generateHeadersForBasicAuth,
    generateHashForRegistering } from '../src/networking';
import { localhost,
    loginText,
    registerText,
    widthWithThirtyPercentPadding} from '../src/static/constants';

export class Login extends Component {
    constructor (props) {
        super(props);
        this.state = {
            page: 'Login',
            username: '',
            password: '',
            email: '',
            textFocused: false
        };
    }

    togglePage () {
        let currentPage = this.getPageTitle(this.state);
        this.setState({ page: currentPage });
    }

    getPageTitle (state) {
        let { page } = state;
        return (page === 'Login') ? 'Sign up' : 'Login';
    }

    getHelperText (state) {
        let { page } = state;
        return (page === 'Login') ? loginText : registerText;
    }

    handleButtonPress(state) {
        const { page, username, password, email } = state;
        if(username === '' || password === '' || (page === 'Sign up' && email === ''))
        {
            Alert.alert('Invalid values', 'Please enter all the values.');
            return;
        }
        if (page === 'Login') {
            this.handleLoginRequest(state)
        } else {
            this.handleRegisteringRequest(state)
        }
    }

    handleRegisteringRequest (state) {
        const { username, password, email } = state;
        let ip_addr = `http://${localhost}:5000/api/v1.0/register`;
        fetch(ip_addr, {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                'user_info': generateHashForRegistering(username, password, email)
            })
        })
        .then((response) => response.json())
        .then((responseJson) => {
            if (responseJson.user_creation === 'success') {
                Alert.alert('User creation', 'User created successfully!');
                console.log(responseJson);
                this.setState({ page: 'Login' });
            } else {
                Alert.alert('User creation', 'User exists already!');
            }
        })
        .catch((error) => {
            console.log('error ', error);
            console.error(error);
        });
    }

    handleLoginRequest (state) {
        let ip_addr = `http://${localhost}:5000/api/v1.0/auth`;
        const { username, password } = state;
        fetch(ip_addr, {
            method: 'GET',
            headers: generateHeadersForBasicAuth(username, password)
        })
        .then((response) => response.json())
        .then((responseJson) => {
            console.log(responseJson);
            if (responseJson.username === username) {
                this.props.onLogin(username, responseJson.token);
            } else {
                Alert.alert('Try again, mate!', 'Invalid credentials.');
                this.setState({ password: '' });
            }
        })
        .catch((error) => {
            console.error(error);
        });
    }

    renderEmailInput() {
        if (this.state.page === 'Sign up') {
            return (
                <TextInput
                    placeholder='Email address'
                    autoCapitalize='none'
                    autoCorrect={false}
                    autoFocus={false}
                    keyboardType='email-address'
                    value={this.state.email}
                    onFocus={() => this.setState({textFocused: true})}
                    onChangeText={(text) => this.setState({ email: text })}
                />
            )
        }
    }

    renderLogoAndWelcomeText() {
        if (this.state.textFocused === false) {
            return (
                <View style={[styles.logoContainer]}>
                  <Text style={[styles.welcomeText]}>Welcome to NearBuy</Text>
                  <Image
                      resizeMode="contain"
                      style={styles.logo}
                      source={require('./../logo.png')}
                  />
                </View>
            )
        }
    }

    render () {
        return (
            <View style={[styles.container]}>
              <KeyboardAvoidingView
                  behavior='padding'
                  keyboardVerticalOffset={64}
              >
                  {this.renderLogoAndWelcomeText()}
                <View style={[styles.loginContainer]}>
                    {this.renderEmailInput()}
                  <View style={{margin: 4}}></View>
                  <TextInput
                      placeholder='Username'
                      autoCapitalize='none'
                      autoCorrect={false}
                      autoFocus={false}
                      onFocus={() => this.setState({textFocused: true})}
                      keyboardType='email-address'
                      value={this.state.username}
                      onChangeText={(text) => this.setState({ username: text })} />
                  <View style={{margin: 4}}></View>
                  <TextInput
                      placeholder='Password'
                      autoCapitalize='none'
                      autoCorrect={false}
                      secureTextEntry={true}
                      onFocus={() => this.setState({textFocused: true})}
                      value={this.state.password}
                      onChangeText={(text) => this.setState({ password: text })} />

                  <Text style={{fontSize: 27}}>{this.state.route}</Text>
                  <Button onPress={() => this.handleButtonPress(this.state)} title={this.state.page}/>
                  <View style={{marginTop: 40, flexDirection: 'row', justifyContent: 'center'}}>
                    <Text onPress={() => this.togglePage()} style={{fontSize: 16, color: 'blue'}}>
                        {this.getPageTitle(this.state)}
                    </Text>
                  </View>
                  <Text style={[styles.loginHint]}>{this.getHelperText(this.state)}</Text>
                </View>
              </KeyboardAvoidingView>
            </View>
        );
    }
}


const mapStateToProps = (state, ownProps) => {
    return {
        isLoggedIn: state.auth.isLoggedIn
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        onLogin: (username, token) => { dispatch(login(username, token)); },
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'space-around',
    },
    welcomeText: {
        fontSize: 24,
        textAlign: 'center',
        marginTop: 50
    },
    loginHint: {
        fontSize: 12,
        textAlign: 'center',
        marginTop: 10,
    },
    logo: {
        width: 350,
        height: 150,
    },
    loginContainer: {
        justifyContent: 'center',
        alignSelf: 'center',
        width: widthWithThirtyPercentPadding,
    },
    logoContainer: {
        justifyContent: 'center',
        alignSelf: 'stretch',
        padding: 0
    },
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
