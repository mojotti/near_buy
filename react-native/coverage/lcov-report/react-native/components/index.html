<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for react-native/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> react-native/components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.57% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>37/89</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.54% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>16/26</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">29.79% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>14/47</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.57% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>37/89</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Items.js"><a href="Items.js.html">Items.js</a></td>
	<td data-value="41.18" class="pic low"><div class="chart"><div class="cover-fill" style="width: 41%;"></div><div class="cover-empty" style="width:59%;"></div></div></td>
	<td data-value="41.18" class="pct low">41.18%</td>
	<td data-value="34" class="abs low">14/34</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="30" class="pct low">30%</td>
	<td data-value="20" class="abs low">6/20</td>
	<td data-value="41.18" class="pct low">41.18%</td>
	<td data-value="34" class="abs low">14/34</td>
	</tr>

<tr>
	<td class="file low" data-value="Login.js"><a href="Login.js.html">Login.js</a></td>
	<td data-value="41.82" class="pic low"><div class="chart"><div class="cover-fill" style="width: 41%;"></div><div class="cover-empty" style="width:59%;"></div></div></td>
	<td data-value="41.82" class="pct low">41.82%</td>
	<td data-value="55" class="abs low">23/55</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="20" class="abs medium">15/20</td>
	<td data-value="29.63" class="pct low">29.63%</td>
	<td data-value="27" class="abs low">8/27</td>
	<td data-value="41.82" class="pct low">41.82%</td>
	<td data-value="55" class="abs low">23/55</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Oct 18 2017 21:32:17 GMT+0300 (EEST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
