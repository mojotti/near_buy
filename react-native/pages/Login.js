'use strict';

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Alert, ScrollView, Text, TextInput, View, Button, Platform, StyleSheet,
  Image, KeyboardAvoidingView } from 'react-native';
import { login } from '../redux/actions/auth';

const LOCALHOST = (Platform.OS === 'ios') ? 'localhost' : '10.0.2.2';
const base64 = require('base-64');
const loginText = "New user? Press 'Sign up' to register.";
const registerText = "Existing user? Press 'Login'.";


class Login extends Component {
    constructor (props) {
        super(props);
        this.state = {
            page: 'Login',
            username: '',
            password: ''
        };
    }

    userLogin (e) {
        this.props.onLogin(this.state.username, this.state.password);
        e.preventDefault();
    }

    togglePage (e) {
      this.setState({ page: this.altLoginSignup });
      e.preventDefault();
    }

    get altLoginSignup () { return (this.state.page === 'Login')
          ? 'Sign up' : 'Login'; }

    get altHelperText () { return (this.state.page === 'Login')
          ? loginText : registerText; }

    getHeaders() {
      var headers = new Headers();
      var hash = this.getHash();
      headers.append("Authorization", "Basic " + hash);
      return headers;
    }
    
    getHash() {
      var credentials = this.state.username + ':' + this.state.password
      return base64.encode(credentials);
    }

    handleLoginRequest (e) {
      e.preventDefault();
      var headers = this.getHeaders();

      fetch('http://' + LOCALHOST + ':5000/api/v1.0/auth', {
         method: 'GET',
         headers: this.getHeaders()
      })

      .then((response) => response.json())
      .then((responseJson) => {
        console.log(responseJson);
        if (responseJson.login === 'success') {
          this.props.onLogin(this.state.username, this.getHash());
        } else {
          Alert.alert('Try again, mate!', 'Invalid credentials.');
          this.setState({ password: '' });
        }
      })

      .catch((error) => {
         console.error(error);
         return;
      });
    }

    render () {
      return (
          <View style={[styles.container]}>
            <KeyboardAvoidingView
            behavior='padding'
            keyboardVerticalOffset={-70}
            >
            <View style={[styles.logoContainer]}>
              <Text style={[styles.welcomeText]}>Welcome to NearBuy</Text>
              <Image
                resizeMode="contain"
                style={styles.logo}
                source={require('./../logo.png')}
              />
            </View>

            <View style={[styles.loginContainer]}>
            <TextInput
                placeholder='Username'
                autoCapitalize='none'
                autoCorrect={false}
                autoFocus={false}
                keyboardType='email-address'
                value={this.state.username}
                onChangeText={(text) => this.setState({ username: text })} />
            <View style={{margin: 4}}/>
            <TextInput
                placeholder='Password'
                autoCapitalize='none'
                autoCorrect={false}
                secureTextEntry={true}
                value={this.state.password}
                onChangeText={(text) => this.setState({ password: text })} />

              <Text style={{fontSize: 27}}>{this.state.route}</Text>
              <Button onPress={(e) => this.handleLoginRequest(e)} title={this.state.page}/>
              <View style={{marginTop: 55, flexDirection: 'row', justifyContent: 'center'}}>
                <Text onPress={(e) => this.togglePage(e)} style={{fontSize: 14, color: 'blue'}}>
                    {this.altLoginSignup}
                </Text>
              </View>
              <Text style={[styles.loginHint]}>{this.altHelperText}</Text>
          </View>
          </KeyboardAvoidingView>
        </View>
    );
  }
}


const mapStateToProps = (state, ownProps) => {
    return {
        isLoggedIn: state.auth.isLoggedIn
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        onLogin: (username, hash) => { dispatch(login(username, hash)); },
        onSignUp: (username, password) => { dispatch(signup(username, password)); }
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  welcomeText: {
    fontSize: 24,
    textAlign: 'center',
    marginTop: 50
  },
  loginHint: {
    fontSize: 12,
    textAlign: 'center',
    marginTop: 10,
  },
  logo: {
    width: 350,
    height: 150,
  },
  loginContainer: {
    justifyContent: 'center',
    alignSelf: 'stretch',
    padding: 80
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
