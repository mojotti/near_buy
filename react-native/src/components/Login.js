import React from 'react';
import { connect } from 'react-redux';
import {
  Alert,
  Image,
  Keyboard,
  KeyboardAvoidingView,
  Platform,
  Text,
  TextInput,
  TouchableHighlight,
  View,
} from 'react-native';
import { login } from '../../redux/actions/auth';
import {
  generateHeadersForBasicAuth,
  getHeadersForRegistering,
  generateHashForRegistering,
} from '../networking/networking';
import {
  localhost,
  loginText,
  logo,
  networkErrorAlert,
  registerText,
} from '../static/constants';
import { styles } from '../static/styles/LoginStyles';

const LOGO = logo;

export class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      page: 'Login',
      username: '',
      password: '',
      email: '',
      isKeyboardVisible: false,
    };

    this.keyboardDidHide = this.keyboardDidHide.bind(this);
    this.keyboardDidShow = this.keyboardDidShow.bind(this);
    this.handleLoginRequest = this.handleLoginRequest.bind(this);
    this.handleRegisteringRequest = this.handleRegisteringRequest.bind(this);
    this.handleRegisteringResponse = this.handleRegisteringResponse.bind(this);
    this.handleError = this.handleError.bind(this);
  }

  componentDidMount() {
    this.keyboardDidShowListener = Keyboard.addListener(
      'keyboardDidShow',
      this.keyboardDidShow,
    );
    this.keyboardDidHideListener = Keyboard.addListener(
      'keyboardDidHide',
      this.keyboardDidHide,
    );
  }

  componentWillUnmount() {
    this.keyboardDidShowListener.remove();
    this.keyboardDidHideListener.remove();
  }

  getAlternativePageTitle() {
    const { page } = this.state;
    return page === 'Login' ? 'Sign up' : 'Login';
  }

  getHelperText() {
    const { page } = this.state;
    return page === 'Login' ? loginText : registerText;
  }

  keyboardDidShow() {
    this.setState({ isKeyboardVisible: true });
  }

  keyboardDidHide() {
    this.setState({ isKeyboardVisible: false });
  }

  togglePage() {
    const alternativePage = this.getAlternativePageTitle();
    this.setState({ page: alternativePage });
  }

  handleButtonPress() {
    const { page, username, password, email } = this.state;
    const hasInvalidUserDetails =
      username === '' ||
      password === '' ||
      (page === 'Sign up' && email === '');

    if (hasInvalidUserDetails) {
      Alert.alert('Invalid values', 'Please enter all the values.');
      return;
    }
    if (page === 'Login') {
      this.handleLoginRequest();
    } else {
      this.handleRegisteringRequest();
    }
  }

  handleRegisteringRequest() {
    const { username, password, email } = this.state;
    const ipAddress = `http://${localhost}:5000/api/v1.0/register`;

    fetch(ipAddress, {
      method: 'POST',
      headers: getHeadersForRegistering(),
      body: JSON.stringify({
        user_info: generateHashForRegistering(username, password, email),
      }),
    })
      .then(response => response.json())
      .then(responseJson => {
        this.handleRegisteringResponse(responseJson);
      })
      .catch(error => {
        this.handleError(error);
      });
  }

  handleRegisteringResponse(response) {
    if (response.user_creation === 'success') {
      Alert.alert('User creation', 'User created successfully!');
      this.setState({ page: 'Login' });
    } else {
      Alert.alert('User creation', 'User exists already!');
    }
  }

  handleError(error) {
    if (error.message === 'Network request failed') {
      Alert.alert('Oops!', networkErrorAlert);
    }
  }

  handleLoginRequest() {
    const ipAddress = `http://${localhost}:5000/api/v1.0/auth`;
    const { username, password } = this.state;
    fetch(ipAddress, {
      method: 'GET',
      headers: generateHeadersForBasicAuth(username, password),
    })
      .then(response => response.json())
      .then(responseJson => {
        this.handleLoginResponse(responseJson);
      })
      .catch(error => {
        this.handleError(error);
      });
  }

  handleLoginResponse(response) {
    if (response.username === this.state.username) {
      this.props.onLogin(this.state.username, response.token);
    } else {
      Alert.alert('Try again, mate!', 'Invalid credentials.');
      this.setState({ password: '' });
    }
  }

  renderEmailInput() {
    if (this.state.page === 'Sign up') {
      return (
        <TextInput
          id="emailTextInput"
          placeholder="Email address"
          style={styles.textInputStyleNegativeMargin}
          autoCapitalize="none"
          autoCorrect={false}
          autoFocus={false}
          keyboardType="email-address"
          value={this.state.email}
          onChangeText={text => this.setState({ email: text })}
          underlineColorAndroid="transparent"
        />
      );
    }
    return null;
  }

  renderLogoAndWelcomeText() {
    if (!this.state.isKeyboardVisible) {
      return (
        <View style={styles.logoContainer}>
          <Text style={styles.welcomeText}>Hi! Welcome to</Text>
          <Image resizeMode="contain" style={styles.logo} source={LOGO} />
        </View>
      );
    }
    return null;
  }

  render() {
    const behavior = Platform.OS === 'ios' ? 'padding' : null;
    return (
      <View style={[styles.container]}>
        <KeyboardAvoidingView behavior={behavior} keyboardVerticalOffset={64}>
          {this.renderLogoAndWelcomeText()}
          <View style={styles.loginContainer}>
            {this.renderEmailInput()}
            <TextInput
              id="usernameTextInput"
              placeholder="Username"
              style={styles.textInputStyleNegativeMargin}
              autoCapitalize="none"
              autoCorrect={false}
              autoFocus={false}
              value={this.state.username}
              underlineColorAndroid="transparent"
              onChangeText={text => this.setState({ username: text })}
            />
            <TextInput
              id="passwordTextInput"
              placeholder="Password"
              autoCapitalize="none"
              style={styles.textInputStyleLargeMargin}
              autoCorrect={false}
              secureTextEntry={true}
              underlineColorAndroid="transparent"
              value={this.state.password}
              onChangeText={text => this.setState({ password: text })}
            />

            <TouchableHighlight
              id="loginButton"
              onPress={() => this.handleButtonPress()}
              style={styles.loginButton}
            >
              <Text style={styles.loginText}>{this.state.page}</Text>
            </TouchableHighlight>
            <View style={styles.altPageContainer}>
              <Text
                id="altPageTitle"
                onPress={() => this.togglePage()}
                style={styles.altPageTitle}
              >
                {this.getAlternativePageTitle()}
              </Text>
            </View>
            <Text id="helperText" style={styles.loginHint}>
              {this.getHelperText()}
            </Text>
          </View>
        </KeyboardAvoidingView>
      </View>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    isLoggedIn: state.authorizationReducer.isLoggedIn,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onLogin: (username, token) => {
      dispatch(login(username, token));
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
