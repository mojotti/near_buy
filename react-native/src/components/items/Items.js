'use strict';

import React from 'react';
import { Button, ListView, Text, View } from 'react-native';
import { logout } from '../../redux/actions/AuthorizationAction';
import { connect } from 'react-redux';
import { localhost, userHasNoItemsText } from '../../static/constants';
import { ListViewItem } from './ListViewItem';
import { styles } from '../../static/styles/ItemsStyles';

export class Items extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      dataSource: new ListView.DataSource({
        rowHasChanged: (row1, row2) => row1 !== row2,
      }),
      loaded: false,
      data: '',
    };
  }

  static navigationOptions = ({ navigation }) => {
    const { navigate } = navigation;
    return {
      title: 'Your items',
      headerLeft: null,
      headerRight: (
        <Text onPress={() => navigate('NewItem')} style={styles.newItem}>
          New item
        </Text>
      ),
    };
  };

  componentDidMount() {
    this.fetchData();
  }

  getHeaders() {
    const headers = new Headers();
    headers.append('Authorization', `Bearer ${this.props.token}`);
    return headers;
  }

  userLogout() {
    this.props.onLogout();
  }

  fetchData() {
    let url = `http://${localhost}:5000/api/v1.0/user/items`;
    fetch(url, {
      method: 'GET',
      headers: this.getHeaders(),
    })
      .then(response => response.json())
      .then(responseJson => {
        this.handleAllItemsRequest(responseJson);
      })
      .catch(error => {
        console.error(error);
      });
  }

  handleAllItemsRequest(responseJson) {
    if (responseJson.items === 'no items') {
      this.setState({
        loaded: true,
        data: responseJson.items,
      });
    } else {
      this.setState({
        dataSource: this.state.dataSource.cloneWithRows(responseJson.items),
        loaded: true,
        data: responseJson.items,
      });
    }
  }

  renderUserData() {
    if (this.state.loaded === false) {
      return <Text style={[styles.infoText]}>{'Loading user data...'}</Text>;
    }
    if (this.state.data === 'no items') {
      return <Text style={[styles.infoText]}>{userHasNoItemsText}</Text>;
    } else {
      return (
        <ListView
          dataSource={this.state.dataSource}
          renderRow={data => <ListViewItem item={data} />}
        />
      );
    }
  }

  render() {
    return (
      <View style={[styles.container]}>
        {this.renderUserData()}
        <Button
          onPress={() => this.userLogout()}
          title="Logout"
          style={{ margin: 15 }}
        />
      </View>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    username: state.authorizationReducer.username,
    token: state.authorizationReducer.token,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onLogout: () => {
      dispatch(logout());
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Items);
