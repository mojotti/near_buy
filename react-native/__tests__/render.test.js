import React from 'react';
import App from './../App';
import { Login } from '../components/Login';
import { Items } from '../components/Items';
import renderer from 'react-test-renderer';
import "isomorphic-fetch";  // for headers, fetch, etc.


it('Login renders without crashing', () => {
  const rendered = renderer.create(
    <Login />
  ).toJSON();
  expect(rendered).toBeTruthy();
});

it('Login renders correctly', () => {
  const tree = renderer.create(
    <Login />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Items renders without crashing', () => {
  const rendered = renderer.create(
    <Items />
  ).toJSON();
  expect(rendered).toBeTruthy();
});

it('Items renders correctly', () => {
  const tree = renderer.create(
    <Items />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
