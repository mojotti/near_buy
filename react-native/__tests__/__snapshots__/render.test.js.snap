// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Items renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "padding": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    Loading user data...
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Logout
      </Text>
    </View>
  </View>
</View>
`;

exports[`Render Login renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
      },
    ]
  }
>
  <View
    keyboardVerticalOffset={64}
    onLayout={[Function]}
    style={
      Array [
        undefined,
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
          "padding": 0,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 24,
            "marginTop": 50,
            "textAlign": "center",
          }
        }
      >
        Welcome to NearBuy
      </Text>
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/static/images/logo.png",
          }
        }
        style={
          Object {
            "height": 150,
            "width": 350,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "justifyContent": "center",
          "width": 525,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        autoFocus={false}
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="Username"
        style={
          Object {
            "marginBottom": 4,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          Object {
            "marginBottom": 4,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 27,
          }
        }
      />
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 40,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          id="altPageTitle"
          onPress={[Function]}
          style={
            Object {
              "color": "blue",
              "fontSize": 16,
            }
          }
        >
          Sign up
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        id="helperText"
        style={
          Object {
            "fontSize": 12,
            "marginTop": 10,
            "textAlign": "center",
          }
        }
      >
        New user? Press 'Sign up' to register.
      </Text>
    </View>
  </View>
</View>
`;
